src/viam/sdk/components/encoder/server.cpp

```diff
@@ -14,132 +15,57 @@ EncoderServer::EncoderServer(std::shared_ptr<ResourceManager> manager) : Resourc
 ::grpc::Status EncoderServer::GetPosition(
     ::grpc::ServerContext* context,
     const ::viam::component::encoder::v1::GetPositionRequest* request,
-    ::viam::component::encoder::v1::GetPositionResponse* response) {
-    if (!request) {
-        return ::grpc::Status(::grpc::StatusCode::INVALID_ARGUMENT,
-                              "Called [Encoder::GetPosition] without a request");
-    };
-
-    const std::shared_ptr<Resource> rb = resource_manager()->resource(request->name());
-    if (!rb) {
-        return grpc::Status(grpc::UNKNOWN, "resource not found: " + request->name());
-    }
-
-    const std::shared_ptr<Encoder> encoder = std::dynamic_pointer_cast<Encoder>(rb);
-
-    AttributeMap extra;
-    if (request->has_extra()) {
-        extra = struct_to_map(request->extra());
-    }
-
-    const Encoder::position result =
-        encoder->get_position(extra, Encoder::from_proto(request->position_type()));
-    response->set_value(result.value);
-    response->set_position_type(Encoder::to_proto(result.type));
-
-    return ::grpc::Status();
+    ::viam::component::encoder::v1::GetPositionResponse* response) noexcept {
+    return make_service_helper<Encoder>(
+        "EncoderServer::GetPosition", this, request)([&](auto& helper, auto& encoder) {
```

Comment:
I wish there were a way to pull the method name from the request or something; I couldn't find anything clean.

Reply:
Yes, that would be nice, but there really isn't a good way to get the string you want here.